homeassistant/components/jewish_calendar/binary_sensor.py

```diff
@@ -75,32 +71,27 @@ async def async_setup_entry(
     entry = hass.data[DOMAIN][config_entry.entry_id]
 
     async_add_entities(
-        JewishCalendarBinarySensor(config_entry.entry_id, entry, description)
+        JewishCalendarBinarySensor(config_entry, entry, description)
         for description in BINARY_SENSORS
     )
 
 
-class JewishCalendarBinarySensor(BinarySensorEntity):
+class JewishCalendarBinarySensor(JewishCalendarEntity, BinarySensorEntity):
     """Representation of an Jewish Calendar binary sensor."""
 
     _attr_should_poll = False
     _attr_entity_category = EntityCategory.DIAGNOSTIC
+
     entity_description: JewishCalendarBinarySensorEntityDescription
 
     def __init__(
         self,
-        entry_id: str,
+        config_entry: ConfigEntry,
         data: dict[str, Any],
         description: JewishCalendarBinarySensorEntityDescription,
     ) -> None:
         """Initialize the binary sensor."""
-        self.entity_description = description
-        self._attr_name = f"{DEFAULT_NAME} {description.name}"
-        self._attr_unique_id = f"{entry_id}-{description.key}"
-        self._location = data[CONF_LOCATION]
-        self._hebrew = data[CONF_LANGUAGE] == "hebrew"
-        self._candle_lighting_offset = data[CONF_CANDLE_LIGHT_MINUTES]
-        self._havdalah_offset = data[CONF_HAVDALAH_OFFSET_MINUTES]
+        super().__init__(config_entry, data, description)
         self._update_unsub: CALLBACK_TYPE | None = None
```

Comment:
This can be set outside the constructor, after which the whole constructor can be removed